src/movies.js

```diff
@@ -1,25 +1,145 @@
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors.
 // How could you "clean" a bit this array and make it unified (without duplicates)?
-function getAllDirectors(moviesArray) {}
+function getAllDirectors(moviesArray) {
+    return moviesArray.map((movie) => movie.director);
+}
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-function howManyMovies(moviesArray) {}
+function howManyMovies(moviesArray) {
+    let movieCounter = 0;
+    moviesArray.forEach(function (movie) {
+        if (
+            movie.director === "Steven Spielberg" && movie.genre.includes("Drama")
+        ) {
+            movieCounter++;
+        }
+    });
+    return movieCounter;
```

Comment:
```suggestion
return moviesArray.filter(movie => movie.director === "Steven Spielberg" && movie.genre.includes("Drama")).length
```

Lo que has hecho esta genial! pero vamos a usar los metodos de los arrays como el filter, si te ves filtramos dentro y como lo que devuelve el metodo filter es un array, tenemos acceso a su propiedad length directamente! :D